autogpt/llm/chat.py

```diff
@@ -153,6 +146,10 @@ def chat_with_ai(
 
                 # Move to the next most recent message in the full message history
                 next_message_to_add_index -= 1
+            from autogpt.memory_management.summary_memory import (
+                get_newly_trimmed_messages,
+                update_running_summary,
+            )
```

Comment:
please bear with that, otherwise circular dependency.
The fix is to create a agent.summary_memory = SummaryMemory() instance.